src/enigma/config.js

```diff
@@ -1,14 +1,19 @@
-import schema from 'enigma.js/schemas/12.20.0.json';
+import schema from 'enigma.js/schemas/12.170.2.json';
 
 import listCache from './list-cache';
 import layoutCache from './layout-cache';
 
+const searchParams = new URLSearchParams(document.location.search);
+const hostname = searchParams.get('hostname') || 'localhost';
+const port = searchParams.get('port') || '9076';
+const app = searchParams.get('app') || 'enginedata';
+
 const ERR_ABORTED = 15;
-const url = process.env.ENGINE_URL;
 
 const config = {
   schema,
-  createSocket: () => new WebSocket(url || `ws://${location.hostname}:9076/app/${+new Date()}`), // eslint-disable-line
+  app,
+  createSocket: () => new WebSocket(`ws://${hostname}:${port}/app/engineData`),
```

Comment:
My bad, we should use the public enigma.js API, so `createSocket: url => new WebSocket(url)` (as long as the `config.url` value is set...)